src/04-spin-boson-model.md

```diff
@@ -0,0 +1,776 @@
+
+```python
+df = make_df_for_energy_scan("$\Delta E$", -4, 4, 201, two_state.shape[0])
+
+for i, row in df.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df.iloc[i,1:] = evals
+```
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[-0.5,5.5],legend=True, 
+        title="Stationary states ($\omega=1$, $U=0.2$)     (Fig 2)");
+plt.ylabel("Energy");
+```
+
+There are many things to say about what we see in Fig 2.
+
+Two main features are:
+1. the splitting of the levels increases with increasing level number i.e increasing number of bosons. 
+2. not all crossings have been split apart into anti-crossings as we expected - this means some levels don't interact with each other at all. 
+
+On 1. Applying our knowledge from Tutorial 1, we would say that the effective coupling between levels (which is proportional to the level splitting) increases with increasing boson number.
+
+On 2. Upon closer inspection, we can see that the level splittings only occur when $\Delta E  \approx n \omega$ where n is an **odd** integer (we'll come to why we now use $\approx$ instead of = shortly). Although it is hard to see in Fig 2, if we zoom in on $\Delta E \approx 3\omega$ we can indeed see an anti-crossing.
+
+To perform this zoom, it is best to perform a higher resolution scan of $\Delta E$:
+
+```python
+df = make_df_for_energy_scan("$\Delta E$", 2.8, 2.95, 201, two_state.shape[0])
+
+for i, row in df.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df.iloc[i,1:] = evals
+```
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[1.9,2],legend=True, 
+        title="Zoom in of the stationary states ($\omega=1$, $U=0.2$)     (Fig 3)");
+plt.ylabel("Energy");
+
+
+```
+
+<!-- #region -->
+
+
+The level splitting seen in Fig 3 is much smaller than those seen in Fig 2 at the primary resonance ($\Delta E \approx \omega$). We can therefore say that the effective coupling between levels is much less for the non-primary resonances.
+
+Fig 3 also shows us that the location of the resonance is somewhat shifted, i.e. the anti-crossing does not occur when $\Delta E = 3 \omega$ but instead $\Delta E \approx 3\omega$. This shift is known as the [Bloch-Siegert shift](https://en.wikipedia.org/wiki/Bloch-Siegert_shift) (see also [Cohen-Tannoudji](https://iopscience.iop.org/article/10.1088/0022-3700/6/8/007) and [Hagelstein](https://iopscience.iop.org/article/10.1088/0953-4075/41/3/035601)) and arises from the the effect of the interaction energy in the Hamiltonian ($E_{I}$). Specifically, the resonance condition should instead be written as $\Delta E + E_{I} = 3\omega$ and hence the value of $\Delta E$ needed for resonance is somewhat reduced. 
+
```

Comment:
here you widely talk about resonance, but I would spend a sentence, or a paragraph, saying:
With resonance here we mean...

But it is complicated becase you call it resonance because you already know that there is a fast rabi oscillation between the states, but maybe one person that read it don't know it already.

Maybe a sentence explicitely linking again to Tutorial 2 for explanation about what we mean by resonance?

Reply:
I've linked back to Tutorial 2 below Fig 1 already, but I'm going to expand it